### PR TITLE
[O2B-953] Log Creation input fields autofill when needed

### PR DESCRIPTION
**Pull request by AnoeBanoe**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- When creating a log from either the environments or runs page, the existing data autofills in the form

### CODE REVIEW

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Logs/LogsModel.js [hunk 1]
```diff
@@ -95,6 +95,20 @@ export class LogsModel extends Observable {
...
+     */
+    _parseNumbers(input) {
+        // Check if the input is an array
+        if (Array.isArray(input)) {
+            return input; // Return as is if it's already an array
+        }
+        // If it's a string, split by comma, convert each part to an integer, and return the array
+        return input.split(',').map((number) => parseInt(number, 10));
+    }
+
```
[the remark this answers is not in the document]
Comment (AnoeBanoe) [reply lib/public/views/Logs/LogsModel.js:111]: done

File: lib/public/views/Logs/LogsModel.js [hunk 2]
```diff
@@ -108,10 +122,10 @@ export class LogsModel extends Observable {
         const relations = {};
 
         if (runNumbers) {
-            relations.runNumbers = (Array.isArray(runNumbers) ? runNumbers : [runNumbers]).map(parseInt);
+            relations.runNumbers = this._parseNumbers(runNumbers);
```
[the remark this answers is not in the document]
Comment (AnoeBanoe) [reply lib/public/views/Logs/LogsModel.js:125]: done

File: lib/public/views/Runs/Details/runDetailsComponent.js [hunk 3]
```diff
@@ -54,6 +54,13 @@ export const runDetailsComponent = (runDetailsModel, router) => runDetailsModel.
             runNumber: run.runNumber,
         });
 
+        // Link options with runNumber as default, and environmentId or lhcFillNumber if available
+        const linkOptions = {
```
[the remark this answers is not in the document]
Comment (AnoeBanoe) [reply lib/public/views/Runs/Details/runDetailsComponent.js:58]: done

File: lib/public/views/Logs/LogsModel.js [hunk 4]
```diff
@@ -96,36 +96,31 @@ export class LogsModel extends Observable {
     }
 
     /**
-     * Parse the input to an array of numbers
-     * @param {string|string[]} input the input to parse
-     * @return {number[]} the parsed numbers
+     * Parse a comma-separated string of numbers to an array of numbers
+     * @param {string} commaSeparatedString The comma-separated string of numbers to parse
```
Comment (martinboulais) [inline lib/public/views/Logs/LogsModel.js:100]: Maybe `commaSeparatedNumbers` instead?

File: lib/public/views/Logs/LogsModel.js [hunk 5]
```diff
@@ -96,36 +96,31 @@ export class LogsModel extends Observable {
...
         if (runNumbers) {
-            relations.runNumbers = this._parseNumbers(runNumbers);
+            relations.runNumbers = this._parseCommaSeparatedNumbers(runNumbers);
         }
         if (lhcFillNumbers) {
-            relations.lhcFillNumbers = this._parseNumbers(lhcFillNumbers);
+            relations.lhcFillNumbers = this._parseCommaSeparatedNumbers(lhcFillNumbers);
         }
         if (environmentIds) {
             relations.environmentIds = Array.isArray(environmentIds) ? environmentIds : [environmentIds];
```
Comment (martinboulais) [inline lib/public/views/Logs/LogsModel.js:126]: If you change the docblock, this should be changed. If you can't have an array, you should probably actually split the environmentIds here too (no conversion to number though)

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Logs/LogsModel.js [hunk 6]
```diff
@@ -95,26 +95,35 @@ export class LogsModel extends Observable {
...
-            relations.runNumbers = (Array.isArray(runNumbers) ? runNumbers : [runNumbers]).map(parseInt);
+            relations.runNumbers = this._parseCommaSeparatedNumbers(runNumbers);
         }
         if (lhcFillNumbers) {
-            relations.lhcFillNumbers = (Array.isArray(lhcFillNumbers) ? lhcFillNumbers : [lhcFillNumbers]).map(parseInt);
+            relations.lhcFillNumbers = this._parseCommaSeparatedNumbers(lhcFillNumbers);
         }
         if (environmentIds) {
-            relations.environmentIds = Array.isArray(environmentIds) ? environmentIds : [environmentIds];
+            relations.environmentIds = environmentIds.split(',');
```
Comment (martinboulais) [inline lib/public/views/Logs/LogsModel.js:126]:
```suggestion
            relations.environmentIds = environmentIds.split(',').map((environmentId) => environmentId.trim());
```

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Logs/LogsModel.js (hunk 6 above)
Comment (AnoeBanoe) [reply lib/public/views/Logs/LogsModel.js]: done

**Review by martinboulais — APPROVED** [review]

**Pull request merged**